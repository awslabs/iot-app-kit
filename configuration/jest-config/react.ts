export default {
  setupFilesAfterEnv: ['jest-extended/all'],
  maxWorkers: '50%',
  moduleFileExtensions: ['js', 'ts', 'jsx', 'tsx'],
  collectCoverage: true,
  testEnvironment: 'jsdom',
  transform: {
    '.*\\.(tsx?|jsx?)$': [
      '@swc/jest',
      {
        jsc: {
          transform: {
            react: {
              runtime: 'automatic',
            },
          },
        },
      },
    ],
  },
  transformIgnorePatterns: [],
  moduleNameMapper: {
    '\\.(css|scss)$': 'identity-obj-proxy',
  },
};
