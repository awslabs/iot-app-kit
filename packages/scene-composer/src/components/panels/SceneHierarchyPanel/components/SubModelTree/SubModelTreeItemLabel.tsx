import React, { DetailedHTMLProps, FC, LabelHTMLAttributes } from 'react';
import { Button } from '@awsui/components-react';

import VisibilityToggle from '../../../../VisibilityToggle';

interface TreeItemLabelProps extends DetailedHTMLProps<LabelHTMLAttributes<HTMLLabelElement>, HTMLLabelElement> {
  onAdd: () => void;
  onVisibilityToggled?: (show: boolean) => void;
  visible: boolean;
}

const SubModelTreeItemLabel: FC<TreeItemLabelProps> = ({ onAdd, visible, onVisibilityToggled, children, ...props }) => {
  return (
    <span className='tm-scene-node-label' {...props}>
      <p className='tm-scene-node-label-inner'>{children}</p>
      <span className='actions'>
        <Button className='tm-icon-button' iconName='add-plus' variant='inline-icon' onClick={onAdd} />
        <VisibilityToggle className='tm-icon-button' visible={visible} onToggle={onVisibilityToggled} />
      </span>
    </span>
  );
};

SubModelTreeItemLabel.displayName = 'SubModelTreeItemLabel';

export default SubModelTreeItemLabel;
