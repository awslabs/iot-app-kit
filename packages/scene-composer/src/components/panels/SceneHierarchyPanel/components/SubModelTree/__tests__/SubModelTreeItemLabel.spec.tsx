import React from 'react';
import { render } from '@testing-library/react';

import SubModelTreeItemLabel from '../SubModelTreeItemLabel';

describe('SubModelTreeItemLabel', () => {
  it('should render as appropriate', () => {
    const onAdd = jest.fn();

    const { container } = render(
      <SubModelTreeItemLabel visible onAdd={onAdd}>
        Label
      </SubModelTreeItemLabel>,
    );

    expect(container).toMatchSnapshot();
  });
});
