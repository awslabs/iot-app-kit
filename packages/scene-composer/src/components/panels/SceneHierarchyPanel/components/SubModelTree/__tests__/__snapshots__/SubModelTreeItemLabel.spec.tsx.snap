// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubModelTreeItemLabel should render as appropriate 1`] = `
<div>
  <span>
    <p>
      Label
    </p>
    <span
      class="actions"
    >
      <div
        class="tm-icon-button"
        data-mocked="Button"
        iconname="add-plus"
        variant="inline-icon"
      />
      <div
        class="tm-visibility-toggle visible tm-icon-button"
        data-mocked="Button"
        variant="inline-icon"
      >
        <div>
          <svg
            fill="none"
            height="14"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#show_svg__a)"
              fill="#879596"
            >
              <path
                class="show_svg__center-dot"
                d="M8 8.89a2 2 0 1 0 0-4 2 2 0 0 0 0 4Z"
                stroke="#879596"
                stroke-miterlimit="10"
              />
              <path
                d="M7.85 3c2.27 0 4.57 1.31 6.85 3.9-1.07 1.22-3.77 3.88-7 3.88-2.25 0-4.42-1.31-6.46-3.9C2.16 5.69 4.61 3 7.85 3Zm0-1C3.11 2 0 6.89 0 6.89c2.67 3.67 5.33 4.89 7.7 4.89 4.74 0 8.3-4.89 8.3-4.89C13 3.22 10.22 2 7.85 2Z"
              />
            </g>
            <defs>
              <clippath
                id="show_svg__a"
              >
                <path
                  d="M0 0h16v9.78H0z"
                  fill="#fff"
                  transform="translate(0 2)"
                />
              </clippath>
            </defs>
          </svg>
        </div>
      </div>
    </span>
  </span>
</div>
`;
